### PR TITLE
Update scratch bootstrap docs

### LINKED ISSUE #309: Document BootStrap: scratch, potentially provide ADD-like command
## Version of Singularity:

3.1.0-rc2

### Expected behavior

Request is to document the 'scratch' BootStrap.

I realize this is somewhat dangerous/confusing for the new user, so I'm not sure the best way to document it.  I would suggest it is listed with the other BootStrap build sources.  But perhaps describe how it is generally not something you can run.

I would like to replicate something like this Dockerfile:
```
FROM scratch
MAINTAINER The CentOS Project <cloud-ops@centos.org> - ami_creator
ADD centos-6.6-20150304_1234-docker.tar.xz /
```
Of course, I can just build using that as my BootStrap.  But I want to start with the absolute minimum number of files in my container to get me to a worthless shell prompt.

### PR DESCRIPTION
**Pull request by dtrudg**
Add a worked example showing a minimal container.

Fixes: #309

### CODE REVIEW

**Review by ikaneshiro — APPROVED** [review]

File: appendix.rst [hunk 1]
```diff
@@ -853,21 +853,49 @@ Note that differently from the ``docker://`` bootstrap agent both ``docker-daemo
...
+        curl -o /tmp/busybox https://www.busybox.net/downloads/binaries/1.31.0-i686-uclibc/busybox
+        # It needs to be executable
+        chmod +x /tmp/busybox
+
+    %files
+        # Copy from host into empty container
+        /tmp/busybox /bin/sh
+
+    %runscript
+       /bin/sh
```
Comment (vsoch) [inline appendix.rst:883]: That's so cool! Just curious, what is the difference in size between the docker provided busybox and this one?

Comment (vsoch) [reply appendix.rst:883]: Since we mention the smaller one here, it might be cool to say how large the original is. Disregard if it's a silly idea. :)

Comment (dtrudg) [reply appendix.rst:883]: The docker one is 764K in SIF format - so it's not that impressive ;-)

**Comment by vsoch** [comment]
hey @dctrud before I open an issue I thought I could ask you here - should scratch work with multistage builds? I just tried it (using master branch) and got:

```
FATAL:   While performing build: conveyor failed to get: invalid image source: invalid reference format
```

**Comment by vsoch** [comment]
oup nevermind - had a lingering docker from in the file! Sorry for the noise.

**Pull request merged**